contract/contracts/forward_name_resolver/farm_controller.ral

```diff
@@ -0,0 +1,86 @@
+Abstract Contract FarmController() extends Constants(), Events() {
+
+    pub fn getFarmKey() -> ByteVec {
+        return Keys.Farm
+    }
+
+    pub fn getFarm() -> Farm {
+        return Farm(subContractId!(getFarmKey()))
+    }
+
+    @using(preapprovedAssets = true, checkExternalCaller = false)
+    pub fn generateFarm() -> ByteVec {
+        let farmExists = contractExists!(subContractId!(getFarmKey()))
+        assert!(!farmExists, ErrorCodes.FarmAlreadyGenerated)
+        let caller = callerAddress!()
+
+        let (encodeImmutableFields, encodeMutableFields) = Farm.encodeFields!(
+            cropTemplateId,
+            selfContractId!(),
+            farmCollectionUri,
+            renewLength - earliestRenew,
+            0
+        )
+
+        return copyCreateSubContractWithToken!{caller -> ALPH: minimalContractDeposit!()}(
+            getFarmKey(),
+            farmTemplateId,
+            encodeImmutableFields,
+            encodeMutableFields,
+            u256Max!()
+        )
+    }
+
+    fn handleCropRewardTokens(caller: Address, rewardTokenAmount: U256) -> () {
+        let rewardToken = getRewardToken()
+        rewardToken.mintTokens(caller, rewardTokenAmount * 1 alph)
+    }
+
+
+    @using(preapprovedAssets = true, checkExternalCaller = false)
+    pub fn mintCrop(alphAmount: U256) -> ByteVec {
+        let caller = callerAddress!()
+        let alphPerToken = minimalContractDeposit!() + mapEntryDeposit!()
+        let rewardTokenAmount = alphAmount / alphPerToken
+        assert!(rewardTokenAmount * alphPerToken == alphAmount, ErrorCodes.IncorrectFarmInputAmount)
+
+        handleCropRewardTokens(
+            caller,
+            rewardTokenAmount
+        )
+
+        let farm = getFarm()
+        let newCropIndex = farm.createCrop{caller -> ALPH: alphAmount}(
+            caller,
+            rewardTokenAmount
+        )
+        let crop = farm.getCrop(newCropIndex)
+
+        emit CropCreated(
+            newCropIndex,
+            alphAmount,
+            caller,
+            crop.getExpires()
+        )
+        return tokenId!(crop)
+    }
+
+    @using(preapprovedAssets = true)
+    pub fn deleteCrop(nftIndex: U256) -> () {
```

Comment:
Question: does the caller need to return the reward tokens to the contract?

Reply:
No they keep the rewards, Crops have the same behaviour as a multiple Name NFTs just without needing the Name text.